src/V3Hasher.cpp

```diff
@@ -29,38 +31,398 @@ class HasherVisitor final : public AstNVisitor {
+            iterateNull(nodep->typedefp());
+            iterateNull(nodep->refDTypep());
+        });
+    }
+    virtual void visit(AstVoidDType* nodep) override {
+        m_hash = hashNode(nodep, false, true, [=]() {});
+    }
+    virtual void visit(AstEnumDType* nodep) override {
+        m_hash = hashNode(nodep, false, false, [=]() {  //
+            m_hash += nodep->uniqueNum();
+        });
+    }
+
+    //------------------------------------------------------------
+    // AstNodeMath
+    virtual void visit(AstNodeMath* nodep) override {
+        m_hash = hashNode(nodep, true, true, [=]() {});
+    }
+    virtual void visit(AstConst* nodep) override {
+        m_hash = hashNode(nodep, true, true, [=]() {  //
+            m_hash += nodep->num().toHash();
+        });
+    }
+    virtual void visit(AstNullCheck* nodep) override {
+        m_hash = hashNode(nodep, true, true, [=]() {});
+    }
+    virtual void visit(AstCCast* nodep) override {
+        m_hash = hashNode(nodep, true, true, [=]() {  //
+            m_hash += nodep->size();
+        });
+    }
+    virtual void visit(AstVarRef* nodep) override {
+        m_hash = hashNode(nodep, true, true, [=]() {
+            if (nodep->varScopep()) {
+                iterateNull(nodep->varScopep());
+            } else {
+                iterateNull(nodep->varp());
+                m_hash += nodep->hiernameToProt();
             }
-        }
-        // Update what will become the above node's hash
-        m_lowerHash += m_cacheInUser4 ? V3Hash(nodep->user4()) : thisHash;
+        });
+    }
+    virtual void visit(AstVarXRef* nodep) override {
+        m_hash = hashNode(nodep, true, true, [=]() {
+            iterateNull(nodep->varp());
+            m_hash += nodep->dotted();
+        });
+    }
+    virtual void visit(AstMemberSel* nodep) override {
+        m_hash = hashNode(nodep, true, true, [=]() {  //
+            m_hash += nodep->name();
+        });
+    }
+    virtual void visit(AstFScanF* nodep) override {
+        m_hash = hashNode(nodep, true, true, [=]() {  //
+            m_hash += nodep->text();
+        });
+    }
+    virtual void visit(AstSScanF* nodep) override {
+        m_hash = hashNode(nodep, true, true, [=]() {  //
+            m_hash += nodep->text();
+        });
+    }
+    virtual void visit(AstTestPlusArgs* nodep) override {
+        m_hash = hashNode(nodep, true, true, [=]() {  //
+            m_hash += nodep->text();
+        });
+    }
+
+    //------------------------------------------------------------
+    // AstNodeStmt
+    virtual void visit(AstNodeStmt* nodep) override {
+        m_hash = hashNode(nodep, false, true, [=]() {});
+    }
+    virtual void visit(AstNodeText* nodep) override {
+        m_hash = hashNode(nodep, false, true, [=]() {  //
+            m_hash += nodep->text();
+        });
+    }
+    virtual void visit(AstNodeCCall* nodep) override {
+        m_hash = hashNode(nodep, false, true, [=]() { iterateNull(nodep->funcp()); });
```

Comment:
You have // to force-break some lines but not others.  I think the break looks better, so perhaps add them on the iterate ones too?

Reply:
Oops, missed some, will fix!